dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql

```diff
@@ -9,18 +9,44 @@
     {% endif %}
 
     {% set build_sql = bq_insert_overwrite(
-        tmp_relation, target_relation, sql, unique_key, partition_by, partitions, dest_columns, on_schema_change
+        tmp_relation, target_relation, sql, unique_key, partition_by, partitions, dest_columns, on_schema_change, copy_partitions
     ) %}
 
     {{ return(build_sql) }}
 
 {% endmacro %}
 
+{% macro bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) %}
+
+  {% for partition in partitions %}
+    {% if partition_by.granularity == 'hour' %}
+      {% set partition = partition.strftime("%Y%m%d%H") %}
+    {% elif partition_by.granularity == 'day' %}
+      {% set partition = partition.strftime("%Y%m%d") %}
+    {% elif partition_by.granularity == 'month' %}
+      {% set partition = partition.strftime("%Y%m") %}
+    {% elif partition_by.granularity == 'year' %}
+      {% set partition = partition.strftime("%Y") %}
+    {% endif %}
+    {% set tmp_relation_partitioned = api.Relation.create(database=tmp_relation.database, schema=tmp_relation.schema, identifier=tmp_relation.table ~ '$' ~ partition, type=tmp_relation.type) %}
+    {% set target_relation_partitioned = api.Relation.create(database=target_relation.database, schema=target_relation.schema, identifier=target_relation.table ~ '$' ~ partition, type=target_relation.type) %}
+    {% do adapter.copy_table(tmp_relation_partitioned, target_relation_partitioned, "table") %}
+  {% endfor %}
+
+{% endmacro %}
+
 {% macro bq_insert_overwrite(
-    tmp_relation, target_relation, sql, unique_key, partition_by, partitions, dest_columns, tmp_relation_exists
+    tmp_relation, target_relation, sql, unique_key, partition_by, partitions, dest_columns, tmp_relation_exists, copy_partitions
 ) %}
-
   {% if partitions is not none and partitions != [] %} {# static #}
+      {{ bq_static_insert_overwrite(tmp_relation, target_relation, sql, partition_by, partitions, dest_columns, copy_partitions) }}
+  {% else %} {# dynamic #}
+      {{ bq_dynamic_insert_overwrite(tmp_relation, target_relation, sql, unique_key, partition_by, dest_columns, tmp_relation_exists, copy_partitions) }}
+  {% endif %}
+
```

Comment:
Missing an an endmacro: 
```suggestion
{% endmacro %}
```

Reply:
Well spotted, I struggled a bit with the rebase so I'm not surprised I missed something 🙈 
Thanks!